lib/internal/encoding.js

```diff
@@ -32,6 +32,21 @@ function lazyBuffer() {
   return Buffer;
 }
 
+function validateEncoder(obj) {
+  if (obj == null || obj[kEncoder] !== true)
+    throw new errors.TypeError('ERR_INVALID_THIS', 'TextEncoder');
+}
+
+function validateDecoder(obj) {
+  if (obj == null || obj[kDecoder] !== true)
+    throw new errors.TypeError('ERR_INVALID_THIS', 'TextDecoder');
+}
+
+function validateArgument(prop, expected, propName) {
+  if (typeof prop !== expected.toLowerCase())
```

Comment:
Using `toLowerCase` is a performance overhead.

I suggest to either split the validation in two functions or to just keep these validations as they were. You can of course also add a fourth argument.

Reply:
@BridgeAR is this last commit inline with your suggestion?